aea/agent.py

```diff
@@ -199,14 +199,20 @@ def start(self) -> None:
 
         :return: None
         """
+        self._start_setup()
+        self._run_main_loop()
+
+    def _start_setup(self) -> None:
+        """
```

Comment:
Please merge latest develop to get your changes from the other branch (now merged).